### PR TITLE
Rich text Editor: Auto-replace plain text emoticons with emoji

### PR DESCRIPTION
**Pull request by langleyd**
Depends on https://github.com/matrix-org/matrix-rich-text-editor/pull/1009

## What it looks like
https://github.com/user-attachments/assets/a60e841b-5f14-4953-8f12-39b2145b6f70

### CODE REVIEW

File: src/components/views/rooms/wysiwyg_composer/components/PlainTextComposer.tsx [hunk 1]
```diff
@@ -52,6 +53,19 @@ export function PlainTextComposer({
     rightComponent,
     eventRelation,
 }: PlainTextComposerProps): JSX.Element {
+    const [isAutoReplaceEmojiEnabled, setIsAutoReplaceEmojiEnabled] = useState(
+        SettingsStore.getValue<boolean>("MessageComposerInput.autoReplaceEmoji"),
```
Comment (dbkr) [inline src/components/views/rooms/wysiwyg_composer/components/PlainTextComposer.tsx:57]: I believe there's also `useSettingValue` to do the getting + watching + unsubscribing for you

File: src/components/views/rooms/wysiwyg_composer/components/WysiwygComposer.tsx [hunk 2]
```diff
@@ -61,9 +68,26 @@ export const WysiwygComposer = memo(function WysiwygComposer({
     const autocompleteRef = useRef<Autocomplete | null>(null);
 
     const inputEventProcessor = useInputEventProcessor(onSend, autocompleteRef, initialContent, eventRelation);
+
+    const [emojiSuggestions, setEmojiSuggestions] = useState(
+        getEmojiSuggestions(SettingsStore.getValue<boolean>("MessageComposerInput.autoReplaceEmoji")),
```
Comment (dbkr) [inline src/components/views/rooms/wysiwyg_composer/components/WysiwygComposer.tsx:73]: as above

**Review by dbkr — APPROVED** [review]
otherwise looks good!

File: src/components/views/rooms/wysiwyg_composer/components/PlainTextComposer.tsx (hunk 1 above)
Comment (langleyd) [reply src/components/views/rooms/wysiwyg_composer/components/PlainTextComposer.tsx:57]: Ah I actually looked for something as such, obviously not hard enough. 😂 thanks Dave

**Pull request merged**